src/helper/safeSetKeyToMetaObject.ts

```diff
@@ -1,8 +1,10 @@
-import { isValidTransactionKey } from "../models/transaction";
+import { isValidTransactionKey, transactionSchema } from "../models/transaction";
 import { isPrototypeAttribute } from "./isPrototypeAttribute";
 import type { IAnyObject } from "../IAnyObject";
 
+const originalSchema = new Set(Object.keys(transactionSchema.paths));
```

Comment:
I have a bad feeling here. What transactionSchema is could be depending on hoisting and based on the runtime.

Can we have  a new function called getTransactionSchemaKeysSet in transactionts? Something like

```typescript
let usedTransactionSchemaForGetTransactionSchemaKeysSet;
let cachedGetTransactionSchemaKeysSet;

export function getTransactionSchemaKeysSet(): Set<string> {
  if (usedTransactionSchemaForGetTransactionSchemaKeysSet === transactionSchema) {
      return cachedTransactionSchemaKeysSet;
  }
  cachedGetTransactionSchemaKeysSet = new Set(Object.keys(transactionSchema.paths));
  usedTransactionSchemaForGetTransactionSchemaKeysSet = transactionschema;
  return cachedGetTransactionSchemaKeysSet;
}
```

And then we consume that in line 9?

Reply:
we need to validate against the original transactionSchema (please check code in v4), transactionSchema is a const so cant be modified.

Reply:
Scratch that. We have already transactionSchemaKeys, which is properly set when we set the TransactionSchema. Use that instead of using originalSchema.

Reply:
let me checkout the PR.

Reply:
we cant use transactionSchemaKeys (it is updated when we use setTransactionSchema) with custom schemas what we need is to exclude original schema attributes

Reply:
you can test the void issue with this https://gist.github.com/dolcalmi/47567ecba8fe5e8fe5742e10eda6efbe

Reply:
```suggestion
```